### ISSUE TITLE
[EC69] is wrongly detected

### ISSUE THREAD

**Issue by tisonv** [green-code-initiative/ecoCode#348]
**Describe the bug**
EC69 is wrongly reported 

**To Reproduce**
```
string? p = Path.GetDirectoryName(resource.RemotePath);

while (p != null && !p.Equals(@"S:\", StringComparison.OrdinalIgnoreCase))
{
      [ calls to various API with p as param ]

      p = Path.GetDirectoryName(p);
}
```
EC69 is detected on `p.Equals(@"S:\", StringComparison.OrdinalIgnoreCase)`

**Expected behavior**
p mutates at each iteration.
There's no invariant function in the loop.
EC69 should not be fired.

**Screenshots**
No screenshot

**Software Versions**
 - SonarQube Version: 10.6
 - SonarAnalyzer.CSharp: 9.30.0.95878
 - Plugin Version: 1.5.0

**Additional context**
No additional context

**Comment by Djoums** [comment]
Right, the warning should not be triggered in this case, I'll look into it. Using string.Equals doesn't have that problem though.

```csharp
while (!string.Equals(d, @"S:\", StringComparison.OrdinalIgnoreCase))
    d = Path.GetDirectoryName(d);
```

**Comment by Djoums** [comment]
The fix is available in version 1.5.1
